### PR TITLE
MM-41331 : Remove post dot menu from RHS post when not hovered upon

### PR DESCRIPTION
**Pull request by M-ZubairAhmed**
#### Summary
- [x] Added dot menu to dom only when root rhs post is hovered upon
- [x] Added A11y navigational controller to root post

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-41331

#### Related Pull Requests
None

#### Screenshots
https://user-images.githubusercontent.com/17708702/151448728-707c8bb7-ac58-4b7f-9b20-5e043576a94e.mov

#### Release Note

```release-note
Renders dot menu of RHS root post to DOM when only hovered upon. 
```

### CODE REVIEW

File: components/rhs_comment/rhs_comment.jsx [hunk 1]
```diff
@@ -599,16 +599,15 @@ export default class RhsComment extends React.PureComponent {
...
                 ref={this.postRef}
+                role='listitem'
                 id={'rhsPost_' + post.id}
                 tabIndex='-1'
+                post={post}
                 className={`a11y__section ${this.getClassName(post, isSystemMessage, isMeMessage)}`}
                 onClick={this.handlePostClick}
                 onMouseOver={this.setHover}
                 onMouseLeave={this.unsetHover}
-                onFocus={this.handlePostFocus}
```
Comment (M-ZubairAhmed) [inline components/rhs_comment/rhs_comment.jsx:611]: `handlePostFocus` was unused

File: components/rhs_root_post/rhs_root_post.test.jsx [hunk 2]
```diff
@@ -189,18 +187,6 @@ describe('components/RhsRootPost', () => {
         expect(props.actions.markPostAsUnread).not.toHaveBeenCalled();
     });
 
-    test('should pass props correctly to PostFlagIcon', () => {
```
Comment (M-ZubairAhmed) [inline components/rhs_root_post/rhs_root_post.test.jsx:192]: wouldnt need this test any more as the dot menu is only visible on hover state

File: components/rhs_root_post/__snapshots__/rhs_root_post.test.jsx.snap [hunk 3]
```diff
@@ -855,52 +773,6 @@ exports[`components/RhsRootPost should match snapshot when flagged 1`] = `
             postId="id"
           />
         </div>
-        <div
```
Comment (larkox) [inline components/rhs_root_post/__snapshots__/rhs_root_post.test.jsx.snap:858]: Should we have any test that its snapshot includes this piece of HTML?

File: components/rhs_root_post/rhs_root_post.jsx [hunk 4]
```diff
@@ -380,7 +413,7 @@ export default class RhsRootPost extends React.PureComponent {
         }
 
         let dotMenuContainer;
-        if (!isPostDeleted && this.props.post.type !== Constants.PostTypes.FAKE_PARENT_DELETED) {
+        if ((!isPostDeleted && this.props.post.type !== Constants.PostTypes.FAKE_PARENT_DELETED) && (this.state.hover || this.state.dropdownOpened || this.state.showEmojiPicker || this.state.a11yActive)) {
```
Comment (larkox) [inline components/rhs_root_post/rhs_root_post.jsx:416]: Line too long. Maybe should we put the second condition (the or chain) in a variable, so it is also more clear what this second condition is representing?

**Review by larkox — APPROVED** [review]
LGTM, just some minor nipticks.

File: components/rhs_root_post/rhs_root_post.jsx (hunk 4 above)
Comment (M-ZubairAhmed) [reply components/rhs_root_post/rhs_root_post.jsx:416]: I couldn't think of a name so i put it in next line :P

File: components/rhs_root_post/__snapshots__/rhs_root_post.test.jsx.snap (hunk 3 above)
Comment (M-ZubairAhmed) [reply components/rhs_root_post/__snapshots__/rhs_root_post.test.jsx.snap:858]: ✅  yes included a test

**Review by hmhealey — APPROVED** [review]

**Comment by jgilliam17** [comment]
Thanks @M-ZubairAhmed 
Looks good, when manually testing - hovering over posts on the RHS shows the post dot menu, menu opens as expected.
There are some cypress failures related to the PR, see report [here](https://mattermost-cypress-report.s3.amazonaws.com/110413-7d6c54a-onprem-ent-webapp-pr-9718-mattermost/mm-ee-test7d6c54a/mochawesome.html) 
Specifically, 2 channel sidebar test cases (top 2 on the list) and messaging (bottom 3 tests on the failed list). Can you please take a look? Thank you 🙂

**Comment by M-ZubairAhmed** [comment]
@jgilliam17 fixed the failed `messaging` test. Although the  `channel sidebar ` tests passed locally. Could you please try re running the e2es

**Comment by jgilliam17** [comment]
I'll re-run shortly (waiting for green checks)

**Comment by jgilliam17** [comment]
Thanks @M-ZubairAhmed 
[E2E report](https://mattermost-cypress-report.s3.amazonaws.com/145716-c74a2c2-onprem-ent-webapp-pr-9718-mattermost/mm-ee-testc74a2c2/mochawesome.html) looks good, no PR related failures

**Review by jgilliam17 — APPROVED** [review]

**Pull request merged**